packages/flutter_tools/lib/src/test/flutter_web_platform.dart

```diff
@@ -167,6 +193,52 @@ class FlutterWebPlatform extends PlatformPlugin {
     }
   }
 
+  final bool updateGoldens;
+  TestGoldenComparator _testGoldenComparator;
+
+  Future<shelf.Response> _goldenFileHandler(shelf.Request request) async {
+    if (request.url.path.contains('flutter_goldens')) {
+      final Map<String, Object> body = json.decode(await request.readAsString()) as Map<String, Object>;
+      final Uri goldenKey = Uri.parse(body['key'] as String);
+      final Uri testUri = Uri.parse(body['testUri'] as String);
+      final num width = body['width'] as num;
+      final num height = body['height'] as num;
+      Uint8List bytes;
+
+      try {
+        final Runtime browser = Runtime.chrome;
+        final BrowserManager browserManager = await _browserManagerFor(browser);
+        final ChromeTab chromeTab = await browserManager._browser.chromeConnection.getTab((ChromeTab tab) {
+          return tab.url.contains(browserManager._browser.url);
+        });
+        final WipConnection connection = await chromeTab.connect();
+        final WipResponse response = await connection.sendCommand('Page.captureScreenshot', <String, Object>{
+          // Clip the screenshot to include only the element.
```

Comment:
Does this actually clip the screenshot in anyway? What if it is offset from the origin?

Reply:
We are calling `window.render()` before taking the screenshot (in `_matchers_web.dart`), so the element is supposed to be at the origin.

https://github.com/flutter/flutter/pull/45530/files#diff-a337fbcec47632d587865bec77ec07ddR88

Reply:
Can you document that here and where the render is done for posterity?

Reply:
Added comment here, and in `_matches_web.dart` as well.